src/mxcp/server/definitions/endpoints/loader.py

```diff
@@ -16,22 +16,43 @@
 logger = logging.getLogger(__name__)
 
 
-def extract_validation_error(error_msg: str) -> str:
+def extract_validation_error(error: ValidationError | Exception | str) -> str:
     """Extract a concise validation error message from jsonschema error.
 
     Args:
-        error_msg: The mxcp error message from jsonschema
+        error: The ValidationError object, Exception, or error message string
 
     Returns:
-        A concise error message
+        A concise error message with property path when available
     """
-
-    # For type errors
+    # Handle ValidationError objects directly
+    if isinstance(error, ValidationError):
+        # Build property path from absolute_path
+        if error.absolute_path:
+            path_parts = []
+            for part in error.absolute_path:
+                if isinstance(part, str):
+                    path_parts.append(part)
+                else:
+                    path_parts.append(str(part))
+            property_path = ".".join(path_parts)
+            return f"Property '{property_path}': {error.message}"
```

Comment:
I think YAML terminology would call this a “key” rather than a “property” (see [spec reference](https://yaml.org/spec/1.2.2/#mapping)). That said, it might be simplest to just report the path directly without labeling it as “key” or anything else? The context already makes it clear what we mean.